### ISSUE TITLE
A quick question about zero_loss

### ISSUE THREAD

**Issue by hoangcuong2011** [NicGian/text-VAE#4]
Hi there,

First of all - I learnt a lot from your code. many thanks!

Second, I was wondering about the role of zero_loss in your code (e.g. line 218 in https://github.com/NicGian/text_VAE/blob/master/vae_text_s2sloss_kl_weight_V18_github.py)

It is just a tensor with zero values for all. What does it do here as the loss of the full model exactly? I was very confused about this.

Many thanks.

Best,

**Comment by NicGian** [comment]
Hi Hoang,

I’m happy the code was helpful for you

Basically zero_loss is just a fake placeholder loss because anyway at the beginning you need to pass a loss to the model; then the real loss that is used for backpropagation is calculated in the variational_loss

Have a nice day,

Giancarlo

 

Da: Hoang Cuong <notifications@github.com> 
Inviato: martedì 26 febbraio 2019 17:20
A: NicGian/text_VAE <text_VAE@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Oggetto: [NicGian/text_VAE] A quick question about zero_loss (#4)

 

Hi there,

First of all - I learnt a lot from your code. many thanks!

Second, I was wondering about the role of zero_loss in your code (e.g. line 218 in https://github.com/NicGian/text_VAE/blob/master/vae_text_s2sloss_kl_weight_V18_github.py)

It is just a tensor with zero values for all. What does it do here as the loss of the full model exactly? I was very confused about this.

Many thanks.

Best,

**Comment by hoangcuong2011** [comment]
Thank you @NicGian 

My last question: what if I want to have multiple losses, and I want to assign a weight for this fake loss?

Apparently I cannot try something like loss_weights=[weight1, ..., weightK] in model.compile(), because we create a zero-loss fake.

Any idea how to fix this issue?
